### PR TITLE
fix: resolve npm scoped versions

### PR DESCRIPTION
**Pull request by JamesPatrickGill**
### What this does

Previously if this parser saw a version specifier like `npm:something@^1.0.0` this whole specifier was used as the version in the depgraph. This is a bit disingenuous as we should be able to resolve the version this actually gets resolved to using the lockfile. This behavior also causes downstream issues in the Snyk CLI.

Now these versions will attempt to be resolved to just the final version range in the specifier.

### CODE REVIEW

File: lib/dep-graph-builders/npm-lock-v2/index.ts [hunk 1]
```diff
@@ -190,16 +190,23 @@ const getChildNode = (
...
-      overrides,
-    );
+    checkOverrides([...ancestry, { name, version }] as Ancestry[], overrides);
+
+  if (override) {
+    version = override;
+  }
+
+  if (version.startsWith('npm:')) {
+    version = version.split('@').pop() || version;
```
Comment (dotkas) [inline lib/dep-graph-builders/npm-lock-v2/index.ts:204]: Is `@` guaranteed to always exist in that string?

**Review by dotkas — APPROVED** [review]

File: lib/dep-graph-builders/npm-lock-v2/index.ts (hunk 1 above)
Comment (dotkas) [reply lib/dep-graph-builders/npm-lock-v2/index.ts:204]: Ok, it returns undefined. Thought it would throw an error.

Comment (JamesPatrickGill) [reply lib/dep-graph-builders/npm-lock-v2/index.ts:204]: Yes, but if it doesn't you just take the full string anyway

Comment (JamesPatrickGill) [reply lib/dep-graph-builders/npm-lock-v2/index.ts:204]: Can change to a `lastIndexOf` check

Comment (dotkas) [reply lib/dep-graph-builders/npm-lock-v2/index.ts:204]: Nah, it's OK by me at least.

**Pull request merged**